mobile-app/app/screens/AppNavigator/screens/Loans/screens/PaybackLoanScreen.tsx

```diff
@@ -95,710 +130,404 @@ export function PaybackLoanScreen({ navigation, route }: Props): JSX.Element {
-      JSON.stringify(tokens),
-    ],
-  });
-
-  useEffect(() => {
-    if (isFocused) {
-      if (loanTokenAmount.symbol === "DUSD") {
-        dispatch(
-          fetchPrice({
-            client,
-            currency: "USD",
-            token: "DFI",
-          })
-        );
-      } else {
-        dispatch(
-          fetchPrice({
-            client,
-            currency: "USD",
-            token: loanTokenAmount.symbol,
-          })
-        );
-      }
-    }
-  }, [address, blockCount, isFocused]);
-
-  useEffect(() => {
-    client.fee
-      .estimate()
-      .then((f) => setFee(new BigNumber(f)))
-      .catch(logger.error);
-  }, []);
-
-  useEffect(() => {
-    return setIsInputEmpty(new BigNumber(amountToPay).isNaN());
-  }, [amountToPay]);
-
-  const onPaymentTokenSelect = (paymentToken: PaymentTokenProps): void => {
-    setSelectedPaymentToken(paymentToken);
-  };
-
-  const navigateToConfirmScreen = (): void => {
+  const navigateToConfirmScreen = async (): Promise<void> => {
     navigation.navigate({
       name: "ConfirmPaybackLoanScreen",
       params: {
         vault,
-        amountToPayInLoanToken: new BigNumber(amountToPayInLoanToken),
-        amountToPayInPaymentToken: amountToPayInPaymentToken,
-        selectedPaymentTokenBalance: selectedPaymentTokenBalance,
-        loanTokenBalance: loanTokenOutstandingBal,
-        paymentToken: selectedPaymentToken,
-        fee,
         loanTokenAmount,
-        excessAmount: isExcess
-          ? new BigNumber(amountToPayInPaymentToken).minus(
-              outstandingBalanceInPaymentToken
-            )
-          : undefined,
         resultingColRatio,
-        paymentPenalty,
-        ...(isConversionRequired && {
-          conversion: {
-            isConversionRequired,
-            DFIToken,
-            DFIUtxo,
-            conversionAmount,
-          },
-        }),
+        loanTokenActivePriceInUSD,
+        amountToPay: new BigNumber(amountToPay),
+        isPaybackDUSDUsingCollateral: routeParams.isPaybackDUSDUsingCollateral,
       },
       merge: true,
     });
   };
-  const onChangeFromAmount = (amount: string): void => {
-    setAmountToPay(amount);
-  };
-
-  const onSubmit = async (): Promise<void> => {
-    if (
-      !hasSufficientPaymentTokenBalance ||
-      vault === undefined ||
-      hasPendingJob ||
-      hasPendingBroadcastJob
-    ) {
-      return;
-    }
 
-    if (isConversionRequired) {
-      queueConvertTransaction(
-        {
-          mode: "utxosToAccount",
-          amount: conversionAmount,
-        },
-        dispatch,
-        () => {
-          navigateToConfirmScreen();
-        },
-        logger
-      );
+  const onChangeFromAmount = async (
+    amount: string,
+    type: AmountButtonTypes
+  ): Promise<void> => {
+    let toastMessage;
+    if (tokenBalance.lt(amount)) {
+      toastMessage = "Insufficient {{unit}}; max amount entered";
+      setValue("amountToPay", tokenBalance.toFixed(8));
```

Comment:
we should set the amount based on the `type`, currently it is always max regardless which button i tap.